kafka-on-docker/README.md

```diff
@@ -0,0 +1,145 @@
+
+## Running Kafka on Docker
+
+To run Kafka on Docker, first confirm your Docker Desktop is running.  Then execute the following command from the `kafka-on-docker` directory:
+
+```shell
+ docker compose up -d
+```
+
+The `-d` flag runs the docker container in detached mode which is similar to running unix commands in the background with a `&`.  To confirm the container is running run this command:
+```shell
+docker logs broker
+```
+And if everything is running ok you'll see something like this at this at the end of screen output:
+
+```text
+[2024-05-21 17:30:58,752] INFO Awaiting socket connections on broker:29092. (kafka.network.DataPlaneAcceptor)
+[2024-05-21 17:30:58,754] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.DataPlaneAcceptor)
+[2024-05-21 17:30:58,756] INFO [BrokerServer id=1] Waiting for all of the authorizer futures to be completed (kafka.server.BrokerServer)
+[2024-05-21 17:30:58,756] INFO [BrokerServer id=1] Finished waiting for all of the authorizer futures to be completed (kafka.server.BrokerServer)
+[2024-05-21 17:30:58,756] INFO [BrokerServer id=1] Waiting for all of the SocketServer Acceptors to be started (kafka.server.BrokerServer)
+[2024-05-21 17:30:58,756] INFO [BrokerServer id=1] Finished waiting for all of the SocketServer Acceptors to be started (kafka.server.BrokerServer)
+[2024-05-21 17:30:58,756] INFO [BrokerServer id=1] Transition from STARTING to STARTED (kafka.server.BrokerServer)
+[2024-05-21 17:30:58,757] INFO Kafka version: 3.7.0 (org.apache.kafka.common.utils.AppInfoParser)
+[2024-05-21 17:30:58,757] INFO Kafka commitId: 2ae524ed625438c5 (org.apache.kafka.common.utils.AppInfoParser)
+[2024-05-21 17:30:58,757] INFO Kafka startTimeMs: 1716312658757 (org.apache.kafka.common.utils.AppInfoParser)
+[2024-05-21 17:30:58,758] INFO [KafkaRaftServer nodeId=1] Kafka Server started (kafka.server.KafkaRaftServer)
+```
+
+Now let's produce and consume a message! To produce a message, let's open a command terminal on the Kafka container:
+
+```shell
+docker exec -it -w /opt/kafka/bin broker sh
+```
+
+Then create a topic:
+```shell
+./kafka-topics.sh --create --topic my-topic --bootstrap-server broker:29092
+```
+The result of this command should be 
+```text
+Created topic my-topic.
+```
+
+**Important**   
+Take note of the `--bootstrap-server` flag.  Because you're connecting to Kafka *_inside_* the container you use `broker:29092` for the host:port.  If you were to use a client outside the container to connect to Kafka, a producer application running on your laptop for example, you'd use `localhost:9092` as the host:port combo to connect.
```

Comment:
```suggestion
Take note of the `--bootstrap-server` flag.  Because you're connecting to Kafka *_inside_* the container, you use `broker:29092` for the host:port.  If you were to use a client outside the container to connect to Kafka, a producer application running on your laptop for example, you'd use `localhost:9092` instead.
```